dogs-initial/src/main/resources/logback-spring.xml

```diff
@@ -0,0 +1,101 @@
+<?xml version = "1.0" encoding = "UTF-8"?>
+
+        <!--*** Part 1-->
+
+<configuration>
+<appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+    <encoder>
+        <pattern>[%d{yyyy-MM-dd' 'HH:mm:ss.SSS}] [%25.25class] [%thread] [%line] [%-5level] %m%n</pattern>
+    </encoder>
+</appender>
+
+<appender name="FILE" class="ch.qos.logback.core.rolling.RollingFileAppender">
+    <File>D:\Users\Github Repos\java-dogs-logging\var\tmp\mylog.log</File>
+    <append>true</append>
+    <encoder>
+        <pattern>[%d{yyyy-MM-dd' 'HH:mm:ss.SSS}] [%C] [%t] [%L] [%-5p] %m%n</pattern>
+    </encoder>
+
+    <rollingPolicy class="ch.qos.logback.core.rolling.SizeAndTimeBasedRollingPolicy">
+
+        <!-- rollover daily -->
+        <fileNamePattern>D:\Users\Github Repos\java-dogs-logging\var\tmp\mylog.%d{yyyy-MM-dd HH}.%i.txt</fileNamePattern>
+        <maxFileSize>15MB</maxFileSize>
+
+        <!-- keep 30 days' worth of history -->
+        <maxHistory>30</maxHistory>
+        <totalSizeCap>5GB</totalSizeCap>
+    </rollingPolicy>
+</appender>
+
+    <root level="INFO">
+        <appender-ref ref="STDOUT"/>
+    </root>
+
+<!--*** Part 3 below -->
+
+<!--*** Part 2-->
+
+    <root level="INFO">
+        <appender-ref ref="FILE"/>
+        <appender-ref ref="STDOUT"/>
+    </root>
+
+
+
+<!--*** Part 4-->
+
+<appender name="TOMCAT" class="ch.qos.logback.core.rolling.RollingFileAppender">
+    <File>/var/tmp/myTomcatLog.log</File>
+    <append>true</append>
+    <encoder>
+        <pattern>[%d{yyyy-MM-dd'T'HH:mm:ss.sss'Z'}] [%C] [%t] [%L] [%-5p] %m%n</pattern>
+    </encoder>
+
+    <rollingPolicy class="ch.qos.logback.core.rolling.SizeAndTimeBasedRollingPolicy">
+
+        <!-- rollover daily -->
+        <fileNamePattern>D:\Users\Github Repos\java-dogs-logging\var\tmp\myTomcatLog.%d{yyyy-MM-dd HH}.%i.txt</fileNamePattern>
+        <maxFileSize>15MB</maxFileSize>
+
+        <!-- keep 30 days' worth of history -->
+        <maxHistory>30</maxHistory>
+        <totalSizeCap>5GB</totalSizeCap>
+    </rollingPolicy>
+</appender>
+
+<root level="OFF">
+    <appender-ref ref="FILE"/>
+    <appender-ref ref="STDOUT"/>
+</root>
+
+
+<!--*** Part 3-->
+
+<logger name="org.springframework.boot" level="OFF">
+    <appender-ref ref="STDOUT" />
+</logger>
+
+<!--*** Part 3 // these are all names for tomcat-->
+
+<logger name="org.apache.catalina" level="INFO">
+    <appender-ref ref="TOMCAT" />
+    <appender-ref ref="STDOUT" />
+</logger>
+
+<logger name="org.apache.coyote" level="INFO">
+    <appender-ref ref="TOMCAT" />
+    <appender-ref ref="STDOUT" />
+</logger>
+
+<logger name="org.apache.tomcat" level="INFO">
+    <appender-ref ref="TOMCAT" />
+    <appender-ref ref="STDOUT" />
+</logger>
+
+<!--*** Part 5-->
+
+<logger name="com.lambdaschool" level="TRACE">
+    <appender-ref ref="STDOUT" />
+</logger>
```

Comment:
You can create more appender-logger pairs to actually log the contents of your message queues to their own files:
• Set the name attributes of your loggers to the class name of your message listeners (ex.: "com.lambdaschool.dogsinitial.Services.MessageListener") - this will track only the logs in that specific scope (file)
• You could divide up your message listeners into different files so that you can set each message queue to have separate log files
• Don't forget to create the corresponding appenders for your new loggers, with distinct file names